generate-go-collector.bash

```diff
@@ -0,0 +1,7 @@
+#!/bin/env bash
+
+set -e
+
+go get github.com/hashicorp/go-version@v1.6.0
+go run prometheus/gen_go_collector_metrics_set.go
+mv -f go_collector_metrics_* prometheus
```

Comment:
I'm confused why this is needed 🤔

Reply:
I notice it doesn't work without it, I'm just struggling to understand what changed in the past that made this necessary :P